aws/src/main/java/org/apache/iceberg/aws/ManagedHttpClientRegistry.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.aws;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import java.lang.ref.Cleaner;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Supplier;
+import org.apache.iceberg.relocated.com.google.common.annotations.VisibleForTesting;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.http.SdkHttpClient;
+
+/**
+ * A registry that manages the lifecycle of shared HTTP clients for AWS SDK v2. Resources are
+ * cleaned up when garbage collected.
+ */
+public class ManagedHttpClientRegistry {
```

Comment:
Are we sure this is something we want to expose? This really is something specifically configured by us for the AwsClient. I don't know if it's beneficial to expose this concept to consumers of the iceberg AWS module.

Reply:
Hmm, I couldn't find a way around it without needing a refactor. The`ManagedHttpClientRegistry` is needed by classes in both `org.apache.iceberg.aws` and `org.apache.iceberg.aws.s3` by `PrefixedS3Client`. To make this work, I think we need to move the client management from PrefixedS3Client to the `aws` package somehow. I'm open to any ideas here.